src/lib/optimizer/optimizer.cpp

```diff
@@ -204,18 +204,24 @@ void Optimizer::validate_lqp(const std::shared_ptr<AbstractLQPNode>& root_node)
         // outside of the LQP that is currently optimized. For example, if you create a bunch of LQP nodes in the test's
         // SetUp method but these are not used in the current LQP, this can cause this assertion to fail. This is only
         // enforced when rules are executed through the optimizer, not when tests call the rule directly.
-        Assert(all_nodes.contains(output), std::string{"Output `"} + output->description() + "` of node `" +
-                                               node->description() + "` not found in LQP");
+        Assert(nodes_by_lqp[lqp].contains(output), std::string{"Output `"} + output->description() + "` of node `" +
+                                                       node->description() + "` not found in LQP");
+
+        for (const auto& [other_lqp, nodes] : nodes_by_lqp) {
```

Comment:
Add a (4) to the list of tasks done that states that we are checking if nodes are only part of a single LQP only?

Reply:
I'd rather rewrite (1):

> Make sure that all outputs found anywhere in the entire plan are also part of the plan

to

> Make sure that all outputs found in an LQP are also part of the same LQP (excluding subqueries)

Reply:
Yep.